tests/determinism/test

```diff
@@ -0,0 +1,107 @@
+bazel clean
+
+# Build the Scala 3 target, explicitly using Scala 3 toolchain
+echo "Building target with Scala 3 toolchain..."
+bazel build --@rules_scala_annex//rules/scala:scala-toolchain=test_zinc_3 --keep_going --remote_executor= --remote_cache= --disk_cache= :tasty_test_lib 
+
+# Get the generated jar file
+bazel_bin=$(bazel info bazel-bin)
+jar_file="$bazel_bin/determinism/tasty_test_lib.jar"
+
+echo "Extracting TASTy files from build..."
+temp_dir=$(mktemp -d)
+
+cleanup() {
+    exit_code=$?
+    # This will be updated later to handle multiple directories
+    for dir in "${temp_dirs[@]}"; do
+        rm -rf "$dir" 2>/dev/null || true
+    done
+    finish $exit_code
+}
+trap cleanup EXIT
+
+# Extract all .tasty files from the jar
+unzip -j "$jar_file" "*.tasty" -d "$temp_dir" 2>/dev/null || {
+    echo "No .tasty files found in jar, checking if they exist at all..."
+    unzip -l "$jar_file" | grep -i tasty || {
+        echo "ERROR: No TASTy files found in the generated jar"
+        echo "Jar contents:"
+        unzip -l "$jar_file"
+        exit 1
+    }
+}
+
+# Verify TASTy files were generated
+tasty_files=$(find "$temp_dir" -name "*.tasty" | wc -l)
+if [ "$tasty_files" -eq 0 ]; then
+    echo "ERROR: No TASTy files were extracted"
+    exit 1
+fi
+
+echo "SUCCESS: Found $tasty_files TASTy file(s) generated by Scala 3 compiler"
+
+# Show what files were found
+echo "Generated TASTy files:"
+find "$temp_dir" -name "*.tasty" -exec basename {} \;
+
+# Test determinism by rebuilding multiple times
+echo "Testing determinism by rebuilding 5 times..."
+
+# Array to store temp directories for each build
+temp_dirs=("$temp_dir")
+
+# Perform 4 additional builds (we already have one)
+for i in $(seq 2 5); do
+    echo "Build $i/5..."
+    bazel clean --expunge
+    bazel build --@rules_scala_annex//rules/scala:scala-toolchain=test_zinc_3 --keep_going --remote_executor= --remote_cache= --disk_cache= :tasty_test_lib
+    
+    # Create temp directory for this build
+    temp_dir_n=$(mktemp -d)
+    temp_dirs+=("$temp_dir_n")
+    
+    # Extract TASTy files from this build
+    unzip -j "$jar_file" "*.tasty" -d "$temp_dir_n" 2>/dev/null || {
+        echo "ERROR: Failed to extract TASTy files from build $i"
+        exit 1
+    }
+done
+
+# Cleanup function to remove temp directories
+cleanup() {
+    exit_code=$?
+    for dir in "${temp_dirs[@]}"; do
+        rm -rf "$dir" 2>/dev/null || true
+    done
+    finish $exit_code
+}
+
+# Compare all builds against the first one
+echo "Comparing TASTy files across all 5 builds for determinism..."
+all_identical=true
+
+for i in $(seq 2 5); do
+    build_num=$((i-1))
+    echo "Comparing build 1 with build $i..."
+    if ! diff -r "${temp_dirs[0]}" "${temp_dirs[$build_num]}" > /dev/null; then
+        echo "ERROR: TASTy files differ between build 1 and build $i - not deterministic"
+        echo "Differences found:"
+        diff -r "${temp_dirs[0]}" "${temp_dirs[$build_num]}"
```

Comment:
Nit: Instead of re-running `diff`, you may consider storing its output and printing it if it failed.